hoptimator-venice/src/main/java/com/linkedin/hoptimator/venice/VeniceStore.java

```diff
@@ -0,0 +1,62 @@
+package com.linkedin.hoptimator.venice;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.avro.Schema;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.impl.AbstractTable;
+
+import com.linkedin.hoptimator.avro.AvroConverter;
+import com.linkedin.venice.client.schema.StoreSchemaFetcher;
+
+
+/** A batch of records from a Venice store. */
+public class VeniceStore extends AbstractTable {
+
+  private static final String KEY_PREFIX = "KEY_";
+  private final StoreSchemaFetcher storeSchemaFetcher;
+
+  public VeniceStore(StoreSchemaFetcher storeSchemaFetcher) {
+    this.storeSchemaFetcher = storeSchemaFetcher;
+  }
+
+  @Override
+  public RelDataType getRowType(RelDataTypeFactory typeFactory) {
+    Schema keySchema = storeSchemaFetcher.getKeySchema();
+    Schema valueSchema = storeSchemaFetcher.getLatestValueSchema();
+
+    // Modify keySchema fields to contain a prefix of "KEY_" for their name in order to not clash with the value schema
```

Comment:
Another way to do this: have one field `KEY` which is either a simple type (if one key) or a `Row(...)` of all key fields (if more than one key). That would yield `KEY.FOO` instead of `KEY_FOO`.